### ISSUE TITLE
`FinC.ignore_unused()` doesn't work for ignoring the "self" parameter

### ISSUE THREAD

**Issue by rahmant3** [fin-language/fin#45]
I have a Fin class with some functions that do not use class members. In the generated code I got compiler warnings because the "self" parameter wasn't used in the generated code. 
```cs
// Fin implementation
private void helper(u32 var1, u32 var2)
```
```c
// Generated code
void PRIVATE_sample_Class_helper(sample_Class* self, uint32_t var1, uint32_t var2) <-- gcc produces a warning due to self not being used.
```

To ignore the warning, I added this line to my class:
```cs
FinC.ignore_unused(this);
```

But unfortunately, it generated this code, which does not compile because the parameter is called “self” in the generated code:
```c
(void)(this);
```

I wanted it to generate this code:
```c
(void)(self);
```

To workaround this issue, I have used the FinC.EchoToC() function
```cs
FinC.EchoToC("  (void)self"); // Ignore "unused variable" compiler warning in generated code.
```

**Comment by adamfk** [comment]
thanks! Fixed in a2746b4.

**Comment by rahmant3** [comment]
This worked for me! Just a thought, maybe Fin could automatically add that line in the future if it sees that the function doesn't use "this" in the body?
